### PR TITLE
force close mongodb

### PR DESCRIPTION
**Pull request by zhaoyao91**
pass `true` to client.close, otherwise the connect will hang up

### CODE REVIEW

**Comment by zhaoyao91** [comment]
by the time I have to use the workaround below in when register mongo plugin:

```
  // this is a bug of fastify-mongodb
  // see https://github.com/fastify/fastify-mongodb/pull/46
  // after this PR is merged, this hook handler could be removed
  fastify.register(async fastify => {
    fastify.addHook('onClose', () => fastify.mongo.client.close(true))
  })
```

**Comment by mcollina** [comment]
In theory, this should not be needed. Can you please make an example where this is needed, and maybe add a unit test?

I'd prefer we had a `forceClose` parameter into the plugin options, disabled by default.

**Comment by zhaoyao91** [comment]
@mcollina well, if `true` is not passed in, the jest test will not finished since the mongo connection hang up.

I agree with your suggestion about `forceClose` option.

**Comment by mcollina** [comment]
Could you update this PR?

**Comment by zhaoyao91** [comment]
I have updated this PR

**Comment by mcollina** [comment]
@zhaoyao91 can you please merge or rebase on top of master? And add a unit tests and docs for this option?

**Comment by zhaoyao91** [comment]
sorry, but I am busy with other works. If u like this feature, could you please take this issue? thx发自我的华为手机-------- 原始邮件 --------主题：Re: [fastify/fastify-mongodb] force close mongodb (#46)发件人：Matteo Collina 收件人：fastify/fastify-mongodb 抄送：Yao Zhao ,Mention @zhaoyao91 can you please merge or rebase on top of master? And add a unit tests and docs for this option?

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/fastify/fastify-mongodb","title":"fastify/fastify-mongodb","subtitle":"GitHub repository","main_image_url":"https://assets-cdn.github.com/images/email/message_cards/header.png","avatar_image_url":"https://assets-cdn.github.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/fastify/fastify-mongodb"}},"updates":{"snippets":[{"icon":"PERSON","message":"@mcollina in #46: @zhaoyao91 can you please merge or rebase on top of master? And add a unit tests and docs for this option?"}],"action":{"name":"View Pull Request","url":"https://github.com/fastify/fastify-mongodb/pull/46#issuecomment-401706372"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/fastify/fastify-mongodb/pull/46#issuecomment-401706372",
"url": "https://github.com/fastify/fastify-mongodb/pull/46#issuecomment-401706372",
"name": "View Pull Request"
},
"description": "View this Pull Request on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
},
{
"@type": "MessageCard",
"@context": "http://schema.org/extensions",
"hideOriginalBody": "false",
"originator": "AF6C5A86-E920-430C-9C59-A73278B5EFEB",
"title": "Re: [fastify/fastify-mongodb] force close mongodb (#46)",
"sections": [
{
"text": "",
"activityTitle": "**Matteo Collina**",
"activityImage": "https://assets-cdn.github.com/images/email/message_cards/avatar.png",
"activitySubtitle": "@mcollina",
"facts": [

]
}
],
"potentialAction": [
{
"name": "Add a comment",
"@type": "ActionCard",
"inputs": [
{
"isMultiLine": true,
"@type": "TextInput",
"id": "IssueComment",
"isRequired": false
}
],
"actions": [
{
"name": "Comment",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueComment\",\n\"repositoryFullName\": \"fastify/fastify-mongodb\",\n\"issueId\": 46,\n\"IssueComment\": \"{{IssueComment.value}}\"\n}"
}
]
},
{
"name": "Close pull request",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"PullRequestClose\",\n\"repositoryFullName\": \"fastify/fastify-mongodb\",\n\"pullRequestId\": 46\n}"
},
{
"targets": [
{
"os": "default",
"uri": "https://github.com/fastify/fastify-mongodb/pull/46#issuecomment-401706372"
}
],
"@type": "OpenUri",
"name": "View on GitHub"
},
{
"name": "Unsubscribe",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"MuteNotification\",\n\"threadId\": 351479632\n}"
}
],
"themeColor": "26292E"
}
]

**Comment by zhaoyao91** [comment]
I have updated this PR to rebase to the latest master

**Comment by mcollina** [comment]
Can you please add a `forceClose: false` line to the first example of https://github.com/fastify/fastify-mongodb#usage?

**Comment by zhaoyao91** [comment]
done

File: README.md [hunk 1]
```diff
@@ -25,6 +25,10 @@ Add it to your project with `register` and you are done!
 const fastify = require('fastify')()
 
 fastify.register(require('fastify-mongodb'), {
+  // force to close the mongodb connection when app stopped
+  // the default value is false
+  forceClose: true,
```
Comment (allevo) [inline README.md]: I'd like to use `false` there

Comment (zhaoyao91) [reply README.md]: the default value is already false as described in the comment above. so generally, the example would show the non-default case.

**Comment by zhaoyao91** [comment]
It's hard to test. And, this is in fact a feature of mongoDB driver, we just proxy the option to it.

**Pull request merged**